docs/adr/adr-0023-adding-stylelint.md

```diff
@@ -0,0 +1,234 @@
+
+To run and check our styles on our component we created two commands in our `package.json` file who are : 
+* `stylelint` which runs Stylelint and checks for errors 
+* `stylelint:fix` which runs Stylelint and checks for errors and auto-fixes the ones that are auto fixable
+* `stylelint:ci` which runs Stylelint and checks for errors and formats the output for our GitHub actions CI.
+
+The benefits of Stylelint are that we now have a coherent style across our component library code base and use the same format everywhere.
+It also allowed us to get rid of the custom ESLint sort plugin which is done by Stylelint directly from now on. 
+
+## Problems encountered
+
+While we are satisfied with our current configuration we encountered some problems in the process with some rules.
+
+### The whitespace rule
+
+First, we had some problems with the whitespace rule (no-eol-whitespace-rule). The purpose of this rule is not to have whitespace after a line ending (e.g: semicolon or empty lines). The problem with this is that the processor we use targets only the CSS inside template literals which means the backtick wasn't included.
+This leads the code going from this: 
+```javascript
+static get styles () {
+    return [
+      skeletonStyles,
+      // language=CSS
+      css`
+        :host {
+          display: block;
+        }
+      `,
+    ];
+```
+to this:
+```javascript
+static get styles () {
+    return [
+      skeletonStyles,
+      // language=CSS
+      css`
+        :host {
+          display: block;
+        }
+`,
+    ];
+```
+As you can see the ending backtick of the template literal isn't formatted correctly because of this rule as the rule is triggered as an empty line with whitespaces in it.
+As a consequence for now we have deactivated this rule. 
+
+### Selector name case
+
+Another problem was the case on our selectors, indeed by default the `Standard` configuration that we extend wants the selectors to be written as `kebab-case` whereas in our components we also used `kebab--case` and `snake__case`.
+
+To resolve this problem we decided to override these rules (id and class selectors) to match our needs which lead us to have the rule as follows: 
+```json
+{
+   "selector-class-pattern": [
+      "^([a-z][a-z0-9]*)((-{1,2}|_{1,2})[a-z0-9]+)*$",
+      {
+        "message": "Expected class selector to be either kebab-case, kebab--case or snake__case"
+      }
+    ],
+    "selector-id-pattern": [
+      "^([a-z][a-z0-9]*)((-{1,2}|_{1,2})[a-z0-9]+)*$",
+      {
+        "message": "Expected id selector to be either kebab-case, kebab--case or snake__case"
+      }
+    ]
+}
+```
+
+To make it work we just modified the regular expression that is directly provided on the Stylelint website to match our needs and provided a custom message to know precisely what's happening otherwise we would just have had a generic message.
```

Comment:
```suggestion
To make it work, we just modified the regular expression that is directly provided on the Stylelint website to match our needs.
We also provided a custom message to know precisely what's happening, otherwise we would just have had a generic message.
```